massa-versioning/src/versioning.rs

```diff
@@ -934,6 +957,128 @@ impl MipStoreRaw {
+                if bounds.contains(&advance.now) {
+                    key.extend(MIP_STORE_PREFIX.as_bytes().to_vec());
+                    mip_info_ser.serialize(mip_info, &mut key)?;
+                    mip_state_ser.serialize(mip_state, &mut value)?;
+                    match state_id {
+                        ComponentStateTypeId::Active => {
+                            batch.insert(key.clone(), Some(value.clone()));
+                        }
+                        _ => {
+                            versioning_batch.insert(key.clone(), Some(value.clone()));
+                        }
+                    }
+                    key.clear();
+                    value.clear();
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    #[allow(dead_code)]
+    fn extend_from_db(&mut self, db: Arc<RwLock<MassaDB>>) {
+        let mip_info_deser = MipInfoDeserializer::new();
+        let mip_state_deser = MipStateDeserializer::new();
+
+        let db = db.read();
+        let handle = db.db.cf_handle(STATE_CF).expect(CF_ERROR);
+
+        // Get data from state cf handle
+        let mut update_data: BTreeMap<MipInfo, MipState> = Default::default();
+        for (ser_mip_info, ser_mip_state) in
+            db.db.prefix_iterator_cf(handle, MIP_STORE_PREFIX).flatten()
+        {
+            println!("Processing ser entry...");
+
+            // deser
+            // FIXME: Result Result instead of using expect?
+            let (_, mip_info) = mip_info_deser
+                .deserialize::<DeserializeError>(&ser_mip_info[MIP_STORE_PREFIX.len()..])
+                .expect("Mip info deser failed");
+
+            let (_, mip_state) = mip_state_deser
+                .deserialize::<DeserializeError>(&ser_mip_state)
+                .expect("Mip state deser failed");
+
+            update_data.insert(mip_info, mip_state);
+        }
+
+        println!("update_data len: {}", update_data.len());
+
+        // FIXME: stats
+        let store_raw_ = MipStoreRaw {
+            store: update_data,
+            stats: MipStoreStats {
+                config: MipStatsConfig {
+                    block_count_considered: 5,
+                    counters_max: 10,
+                },
+                latest_announcements: Default::default(),
+                network_version_counters: Default::default(),
+            },
+        };
+        let (updated, added) = self
+            .update_with(&store_raw_)
+            .expect("update with state cf data");
+        println!("updated len: {}, added len: {}", updated.len(), added.len());
+
+        let mut update_data: BTreeMap<MipInfo, MipState> = Default::default();
+        let versioning_handle = db.db.cf_handle(VERSIONING_CF).expect(CF_ERROR);
+        // Get data from state cf handle
+        for (ser_mip_info, ser_mip_state) in db
+            .db
+            .prefix_iterator_cf(versioning_handle, MIP_STORE_PREFIX)
```

Comment:
You have to check that the keys starts with the prefix, and break when they do not. I know, it's weird behaviour, but the prefix_iterator keep iterating once the keys don't have the prefix ><